testing/cmocka/CMakeLists.txt

```diff
@@ -43,6 +43,8 @@ if(CONFIG_TESTING_CMOCKA)
         ${CMAKE_CURRENT_LIST_DIR}/0005-cmocka-cmocka_private-fix-warning-in-cmocka_private.patch
         && patch -p0 -d ${CMAKE_CURRENT_LIST_DIR}/cmocka <
         ${CMAKE_CURRENT_LIST_DIR}/0006-fix-linux-risc-v-compile-error-list_initialize.patch
+        && patch -p0 -d ${CMAKE_CURRENT_LIST_DIR}/cmocka <
```

Comment:
but no compiler error happen in cmocka:
https://github.com/apache/nuttx/actions/runs/11834392207/job/32975068050?pr=14697

Reply:
i test in linux, ftruncate is need include <unistd.h>. ftruncate is not use in cmocka mainline

Reply:
let's update to cmocka new version instead